pages/policy-groups/new.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright 2021 The Rode Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState } from "react";
+import { useTheme } from "providers/theme";
+import { useRouter } from "next/router";
+import Button from "components/Button";
+import Input from "components/Input";
+import styles from "styles/modules/PolicyGroupForm.module.scss";
+import PageHeader from "components/layout/PageHeader";
+import { useFormValidation } from "hooks/useFormValidation";
+import { schema } from "schemas/policy-group-form";
+import { showError } from "utils/toast-utils";
+
+const CreateNewPolicyGroup = () => {
+  const { theme } = useTheme();
+  const router = useRouter();
+
+  const [name, setName] = useState("");
+  const [description, setDescription] = useState("");
+  const [loading, setLoading] = useState(false);
+
+  const { isValid, validateField, errors } = useFormValidation(schema);
+
+  const onSubmit = async (event) => {
+    event.preventDefault();
+    const formData = {
+      name,
+      description,
+    };
+
+    const validForm = isValid(formData);
+
+    if (!validForm) {
+      return;
+    }
+
+    setLoading(true);
+    const response = await fetch("/api/policy-groups", {
+      method: "POST",
+      body: JSON.stringify(formData),
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+    setLoading(false);
+
+    if (!response.ok) {
+      showError("Failed to create the policy group.");
+      return;
+    }
+
+    // TODO: once implemented, push to created policy group page
+    router.push("/policy-groups");
+  };
+
+  return (
+    <>
+      <PageHeader>
+        <h1>Create Policy Group</h1>
+      </PageHeader>
+      <div className={`${styles.pageContainer} ${styles[theme]}`}>
+        <form onSubmit={onSubmit} className={styles.form}>
+          <div className={styles.formContentContainer}>
+            <Input
+              name={"name"}
+              label={"Policy Group Name"}
+              placeholder={"ex: pci-bundle"}
+              onChange={(event) => {
+                setName(event.target.value);
+              }}
+              value={name}
+              horizontal
+              onBlur={validateField}
```

Comment:
The form validation on submit works well, but this doesn't seem to be firing after I type an invalid name and tab/mouse over to the description field.

Reply:
The validation hook is set up to only validate onBlur once the form has been submitted once. BUT I do think it would be a good idea to validate this onBlur regardless of the form submission since the name has stricter requirements. I'll update this.